samples/nfc/tag_reader/src/main.c

```diff
@@ -43,6 +57,180 @@ static const struct st25r3911b_nfca_buf rx_buf = {
 	.len = sizeof(rx_data)
 };
 
+enum {
+	T2T_IDLE,
+	T2T_HEADER_READ,
+	T2T_DATA_READ
+};
+
+struct t2t_tag {
+	u32_t state;
+	u16_t data_bytes;
+	u8_t data[NFCA_T2T_BUFFER_SIZE];
+};
+
+static struct t2t_tag t2t;
+
+static int ftd_calculate(u8_t *data, size_t len)
+{
+	u8_t ftd_align;
+
+	ftd_align = (data[len - 1] & NFCA_LAST_BIT_MASK) ?
+		NFCA_FDT_ALIGN_84 : NFCA_FDT_ALIGN_20;
+
+	return len * NFCA_BD * BITS_IN_BYTE + ftd_align;
+}
+
+static int nfc_t2t_read_block_cmd_make(u8_t *tx_data,
+				       size_t tx_data_size,
+				       u8_t block_num)
+{
+	if (!tx_data) {
+		return -EINVAL;
+	}
+
+	if (tx_data_size < NFC_T2T_READ_CMD_LEN) {
+		return -ENOMEM;
+	}
+
+	tx_data[0] = NFC_T2T_READ_CMD;
+	tx_data[1] = block_num;
+
+	return 0;
+}
+
+static int t2t_header_read(void)
+{
+	int err;
+	int ftd;
+
+	err = nfc_t2t_read_block_cmd_make(tx_data, sizeof(tx_data), 0);
+	if (err) {
+		return err;
+	}
+
+	ftd = ftd_calculate(tx_data, sizeof(tx_data));
+
+	t2t.state = T2T_HEADER_READ;
+
+	err = st25r3911b_nfca_transfer_with_crc(&tx_buf, &rx_buf, ftd);
```

Comment:
Can we extend the API of st25r3911b_nfca_transfer_with_crc, so that if we pass NULL pointer instead of ftd, the ftd will be calculated by the driver in the default way based on tx_data?

Reply:
This is simple in case of T2T, when we extend this sample to support T4T then the reader library have to have an automatic RATS command sending. So maybe we can do it in separated PR ?

Reply:
okay